maven-model-builder/src/main/java/org/apache/maven/model/interpolation/AbstractStringBasedModelInterpolator.java

```diff
@@ -85,13 +79,22 @@
     private final PathTranslator pathTranslator;
     private final UrlNormalizer urlNormalizer;
 
+    @Inject
+    private ModelVersionProcessor versionProcessor;
+
     @Inject
     public AbstractStringBasedModelInterpolator( PathTranslator pathTranslator, UrlNormalizer urlNormalizer )
     {
         this.pathTranslator = pathTranslator;
         this.urlNormalizer = urlNormalizer;
     }
 
+    public AbstractStringBasedModelInterpolator setVersionPropertiesProcessor( ModelVersionProcessor processor )
```

Comment:
I do not fully understand why this is a setter, but the private field is injected. Why not inject on directly on the ctor? At the end this is a required field not optional, no?

Reply:
it seems that the tests do not use injection but require to set injected items, e.g. [DefaultModelValidator.java](https://github.com/apache/maven/pull/675/files/614a44f28924cc2bdfb6c668d0004c2086da5267#diff-690df442061d3325c6a7cf7b687e90ae79dd3639d45b8d1e606ad1bbc97a00e2) uses always fields for injection and as an Abstract class it seems a bit strange to inject via ctro

Reply:
Please switch to ctor injection, this will give you two benefits:
1. All required objects are wired and are not optional
2. You can wire manually for testing if necessary, opposite of field injection

Reply:
Done.